src/sentry/rules/conditions/event_frequency.py

```diff
@@ -304,15 +306,17 @@ def query_hook(self, event: Event, start: datetime, end: datetime, environment_i
                 percent_intervals[self.get_option("interval")][1].total_seconds() // 60
             )
             avg_sessions_in_interval = session_count_last_hour / (60 / interval_in_minutes)
+
+            group = event.group if event.group is not None else event.groups[0]
```

Comment:
We shouldn't actually do this check here. We should just rely on `event.group`. Instead, when passing the event to the rule engine in your tests you can use `event.for_group(group)` to get a `GroupEvent`.

We'll need to change the typing hint to accept `Event | GroupEvent`. Moving forward, we actually only want to use `GroupEvent` here, but we can handle that later.

Reply:
Ooh I was wondering if there was a better way to do this, it felt hacky.